src/order/zorn.lean

```diff
@@ -408,3 +414,29 @@ lemma directed_of_chain {α β r} [is_refl β r] {f : α → β} {c : set α}
   (λ hab, (h a ha b hb hab).elim
     (λ h : r (f a) (f b), ⟨⟨b, hb⟩, h, refl _⟩)
     (λ h : r (f b) (f a), ⟨⟨a, ha⟩, refl _, h⟩))
+
+/-- Every chain is contained in a maximal chain. This generalizes Hausdorff's maximality principle.
+-/
+theorem chain.max_chain_of_chain {α r} (c : set α) (hc : zorn.chain r c) :
+  ∃ M, @zorn.is_max_chain _ r M ∧ c ⊆ M :=
+begin
+  obtain ⟨M, ⟨_, hM₀⟩, hM₁, hM₂⟩ := zorn.zorn_subset_nonempty {s | c ⊆ s ∧ zorn.chain r s} _ c
+  ⟨rfl.subset, hc⟩,
+    { refine ⟨M, ⟨hM₀, _⟩, hM₁⟩,
+      rintros ⟨d, hd, hdM₀, hdM₁⟩,
+      have := hM₂ _ _ hdM₀,
+      induction this,
+        { exact hdM₁ hdM₀ },
+      change c ⊆ M with c ≤ M at hM₁,
+      exact ⟨le_trans hM₁ hdM₀, hd⟩ },
+  rintros cs hcs₀ hcs₁ ⟨s, hs⟩,
+  refine ⟨⋃₀ cs, ⟨λ _ ha, set.mem_sUnion_of_mem ((hcs₀ hs).left ha) hs, _⟩,
+    λ _, set.subset_sUnion_of_mem⟩,
+  rintros y ⟨sy, hsy, hysy⟩ z ⟨sz, hsz, hzsz⟩ hyz,
+  by_cases hsseq : sy = sz,
+    { induction hsseq,
+      exact (hcs₀ hsy).right _ hysy _ hzsz hyz, },
+  cases hcs₁ _ hsy _ hsz hsseq with h h,
+    { exact (hcs₀ hsz).right _ (h hysy) _ hzsz hyz },
+    { exact (hcs₀ hsy).right _ hysy _ (h hzsz) hyz }
```

Comment:
```suggestion
  by_cases hsseq : sy = sz,
  { induction hsseq,
    exact (hcs₀ hsy).right _ hysy _ hzsz hyz, },
  cases hcs₁ _ hsy _ hsz hsseq with h h,
  { exact (hcs₀ hsz).right _ (h hysy) _ hzsz hyz },
  { exact (hcs₀ hsy).right _ hysy _ (h hzsz) hyz }
```
the indent linter should complain about this anyway

Reply:
Good to know, I wasn't aware of these indenting guidelines.